app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/mini_game/CoffeeBoostActivity.java

```diff
@@ -0,0 +1,179 @@
+        GridLayout.Spec col1 = GridLayout.spec(0);
+        GridLayout.Spec col2 = GridLayout.spec(1);
+        GridLayout.Spec col3 = GridLayout.spec(2);
+        rows.add(row1); rows.add(row2); rows.add(row3);
+        columns.add(col1); columns.add(col2); columns.add(col3);
+        shuffle();
+    }
+
+    /**
+     * Swap buttons if they are correctly positioned.
+     *
+     * @param clicked
+     *              The clicked button.
+     */
+    private void makeMove(final Button clicked) {
+        int number = Integer.parseInt((String) clicked.getText())-1;
+        int position = current_order.indexOf(number);
+        int empty_space_position = current_order.indexOf(0);
+
+        if (sameRowOrColumn(empty_space_position, position) &&
+                areNeighbours(empty_space_position, position)) {
+
+
+            gridLayout.removeView(buttons[0]);
+            gridLayout.removeView(buttons[number]);
+
+            gridLayout.addView(buttons[0], new GridLayout.LayoutParams(rows.get((position/3)), columns.get((position%3))));
+            gridLayout.addView(buttons[number], new GridLayout.LayoutParams(rows.get((empty_space_position/3)), columns.get((empty_space_position%3))));
+
+            current_order.remove(position);
+            current_order.add(position, 0);
+            current_order.remove(empty_space_position);
+            current_order.add(empty_space_position, number);
+        }
+    }
+
+    /**
+     * Check if the buttons are in the same row or column.
+     *
+     * @param loc1
+     *              Location of the first button.
+     * @param loc2
+     *             Location of the second button.
+     * @return True if they are in the same row or column.
+     */
+    private boolean sameRowOrColumn(int loc1, int loc2) {
+        return ((loc1/3) == (loc2/3) || (loc1%3) == (loc2%3));
+    }
+
+    /**
+     * Check if the tapped button is next to the empty space.
+     *
+     * @param loc1
+     *              Location of the first button.
+     * @param loc2
+     *             Location of the second button.
+     * @return True if the distance is 1 in any direction.
+     */
+    private boolean areNeighbours(int loc1, int loc2) {
+        return ((Math.abs((loc1/3) - (loc2/3)) == 1)
+        || Math.abs((loc1%3) - (loc2%3))  == 1);
+    }
+
+    /**
+     * Check if the game is completed.
+     *
+     * @return True if completed, otherwise false.
+     */
+    private boolean gameComplete() {
+        for(int i = 0; i < 9; i++) {
+            if (!current_order.get(i).equals(original_order[i])) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Check if the game is complete and send a message.
+     *
+     * @param view
+     *          A view.
+     */
+    public void confirmCoffeeEvent(View view) {
+        if (gameComplete()) {
+            game.sendMessage(new EventCompletedMessage(EventTypes.COFFEE_BOOST.ordinal()));
+        }
```

Comment:
Send the user back to the Janitor Activity after pressing the button. This way the player cannot remain the solution and spam ik whenever necessary. When failing the player must try all over again.